### PR TITLE
chore: wire it all together

### LINKED ISSUE #168: Wire all things together: Complete the Order and Position Lifecycle by wiring the systems together
- [ ] Wire the orderbook stream into the app and consume the matching event instead of triggering the trade upon order submission.

### LINKED ISSUE #219: Order matching logic
The orderbook should match a `market order` with the best `limit order`. We only support single matches for the time being.

### PR DESCRIPTION
**Pull request by holzeis**
I leave this PR in draft, as I haven't managed to run a complete test. It might be related to some other issues with the database on my local setup.

- [ ] Testing

resolves #168 
resolves #219

### CODE REVIEW

File: mobile/native/src/trade/order/handler.rs [hunk 1]
```diff
@@ -22,11 +25,19 @@ use trade::Direction;
...
-    // In case we fail to submit to the orderbook we assign our own internal uuid and then return
-    // failure.
+    let url = format!("http://{}", config::get_http_endpoint());
+    let orderbook_client = OrderbookClient::new(Url::parse(&url)?);
+
+    let order_id = match orderbook_client.post_new_order(order.into()).await {
+        Ok(order_response) => order_response.id,
+        Err(e) => {
+            tracing::error!("Failed to post new order. Error: {e:#}");
+            Uuid::new_v4()
```
Comment (1010Tom) [inline mobile/native/src/trade/order/handler.rs:25]: 🙈

I guess this is not on purpose?

File: mobile/native/src/orderbook.rs [hunk 2]
```diff
@@ -22,26 +22,23 @@ pub fn subscribe(secret_key: SecretKey) -> Result<()> {
...
 
-                while let Ok(Some(result)) = stream.try_next().await {
-                    tracing::info!("Received: {result}");
-                }
-            })
-            .await;
+            while let Ok(Some(result)) = stream.try_next().await {
+                tracing::info!("Received: {result}");
+            }
+        });
```
Comment (1010Tom) [inline mobile/native/src/orderbook.rs:70]: nice find

File: crates/orderbook-commons/src/lib.rs [hunk 3]
```diff
@@ -154,6 +154,37 @@ pub struct FilledWith {
...
+    pub fn weighted_execution_price(&self) -> Decimal {
+        if self.matches.len() == 1 {
+            return self
+                .matches
+                .first()
+                .expect("to be exactly one")
+                .execution_price;
+        }
+
+        // TODO: Make sure this is correct
```
Comment (1010Tom) [inline crates/orderbook-commons/src/lib.rs]: Added a fixup for this part.

**Review by 1010Tom — APPROVED** [review]
Works™️

**Review by bonomat — APPROVED** [review]
Tested in a call together with @da-kami / @luckysori and @bonomat

**Pull request merged**